src/spikeinterface/core/recording_tools.py

```diff
@@ -131,8 +131,10 @@ def write_binary_recording(
         data_size_bytes = dtype_size_bytes * num_frames * num_channels
         file_size_bytes = data_size_bytes + byte_offset
 
+        # create a file with file_size_bytes
         file = open(file_path, "wb+")
-        file.truncate(file_size_bytes)
+        file.seek(file_size_bytes - 1)
```

Comment:
```suggestion
        if platform.system() == 'Windows':
            file.seek(file_size_bytes - 1)
            file.write(b"\0")
        else:
            file.truncate(file_size_bytes)
```

We could do something like this. Then we would just need to add `import platform` at the top.

Reply:
No, this would work, did you check on mac, it should posix as well but if you have the chance it would cost nothing.

Reply:
Let me label my checks so we can go back easier and see what I did :) 

Mac was faster than linux. Windows was brutally slow.

Reply:
So to summarize

seek-write is 100x faster on Windows
truncate is 2x faster on Mac and Linux.

Reply:
Yeah, I would like to find out what's up but this is a great speed up! I say let's add your suggestion.